src/EnergyPlus/IndoorGreen.cc

```diff
@@ -0,0 +1,778 @@
+            }
+            lw->getInputFlag = false;
+        }
+        if (lw->NumIndoorGreen > 0) {
+            InitIndoorGreen(state);
+
+            // Simulate evapotranspiration from indoor living walls
+            ETModel(state);
+        } 
+    }
+
+    void GetIndoorGreenInput(EnergyPlusData &state, bool &ErrorsFound)
+    {
+        // PURPOSE OF THIS SUBROUTINE:
+        // Get the input for the indoor living wall objects and store the input data in the indoorgreen array.
+
+        // METHODOLOGY EMPLOYED:
+        // Use the Get routines from the InputProcessor module.
+        auto &lw = state.dataIndoorGreen;
+        auto &ip = state.dataInputProcessing->inputProcessor;
+        static constexpr std::string_view RoutineName("GetIndoorLivingWallInput: ");
+        std::string_view cCurrentModuleObject = "IndoorLivingWall"; //match the idd
+        int NumNums;        // Number of real numbers returned by GetObjectItem
+        int NumAlphas;      // Number of alphanumerics returned by GetObjectItem
+        //int IndoorGreenNum; // Indoor Green index
+        int IOStat;         // Status flag from GetObjectItem
+        Real64 SchMin;
+        Real64 SchMax;
+
+        lw->NumIndoorGreen = ip->getNumObjectsFound(state, cCurrentModuleObject);
+        // Get input for IndoorLivingWall objects
+        if (lw->NumIndoorGreen > 0)
+            lw->indoorgreen.allocate(lw->NumIndoorGreen); // Allocate the IndoorGreen input data array
+        // Input the data for each Indoor Living Wall Object
+        for (int IndoorGreenNum = 1; IndoorGreenNum <= lw->NumIndoorGreen; ++IndoorGreenNum) {
+            ip->getObjectItem(state,
+                                                                     cCurrentModuleObject,
+                                                                     IndoorGreenNum,
+                                                                     state.dataIPShortCut->cAlphaArgs,
+                                                                     NumAlphas,
+                                                                     state.dataIPShortCut->rNumericArgs,
+                                                                     NumNums,
+                                                                     IOStat,
+                                                                     state.dataIPShortCut->lNumericFieldBlanks,
+                                                                     state.dataIPShortCut->lAlphaFieldBlanks,
+                                                                     state.dataIPShortCut->cAlphaFieldNames,
+                                                                     state.dataIPShortCut->cNumericFieldNames);
+            Util::IsNameEmpty(state, state.dataIPShortCut->cAlphaArgs(1), cCurrentModuleObject, ErrorsFound);
+            lw->indoorgreen(IndoorGreenNum).IndoorGreenName = state.dataIPShortCut->cAlphaArgs(1);
```

Comment:
Suggest using `auto &ig = lw->indoorgreen(IndoorGreenNum);` here.  It will help with code readability. Also, suggest renaming array to `indoorGreens`.